department-of-veterans-affairs/vets-website
# Fix ClientError VSP Component not Autofocusing

## Description
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/409

## Testing done
Manual testing using the Apply for Caregiver Assistance form.
All unit tests are passing

To test locally:
1. Add `submission.status=‘clientError’;` above the if statement on line 24 of `src/platform/forms-system/src/js/review/SubmitButtons.jsx`.
2. Run the local mock api `yarn mock-api --responses src/platform/testing/local-dev-mock-api/common.js`
3. Run the app in a different terminal window `yarn watch`
4. Navigate to http://localhost:3001/family-member-benefits/apply-for-caregiver-assistance-form-10-10cg/
5. Fill out the form with whatever data you’d like. You’ll need to go through several screens, clicking "continue" each time. Tip: for the SSNs, the first three digits have to match known patterns. Use the first three digits of a real SSN and then fake the rest.
6. On the last screen, after you click “continue” [double check this], the error message should appear and the screen should auto-scroll to it.

## Screenshots
![error-scroll](https://user-images.githubusercontent.com/12739849/112995295-136ce880-9139-11eb-8000-1379307ccce3.gif)


## Acceptance criteria
- [ ] Scrolling behavior on form has not otherwise changed (each new form page should autoscroll to the top of the form)
- [ ] When an error message is shown on the form, the page should autoscroll to the error
- [ ] When using a screen reader, when the page is loaded with the error, the screen reader focus should go directly to the error.

## Definition of done
